lib/active_merchant/billing/gateways/stripe.rb

```diff
@@ -459,6 +460,20 @@ def non_fractional_currency?(currency)
       def emv_payment?(payment)
         payment.respond_to?(:emv?) && payment.emv?
       end
+
+      def error_code_from(response)
```

Comment:
I can use parts of your suggestion to simplify what I have though. Change coming shortly once I test it.